index.d.ts

```diff
@@ -0,0 +1,84 @@
+import * as fastify from "fastify";
+import * as http from "http";
+import { HttpErrors } from "./lib/httpError";
+
+declare module "fastify" {
+  namespace SensibleTypes {
+    type ToType<T> = [Error, T];
+    type Address = 'loopback' | 'linklocal' | 'uniquelocal' | string;
+  }
+
+  interface Assert {
+    (condition: boolean, code?: number | string, message?: string): string;
+    ok(condition: boolean, code?: number | string, message?: string): string;
+    equal<T, U>(a: T, b: U, code?: number | string, message?: string): string;
```

Comment:
Could the equality expressions be simplified so that the `a` and `b` parameters are the same generic type? If something is to be equivalent I believe they would also have the same type

Reply:
I do agree for most equality expressions, however I'd maintain different generics for `strictEqual()`, `notStrictEqual`, `deepEqual` and `notDeepEqual` which will check for the type as well, thus, the objects can be different. Does it make sense?

Reply:
I based the types test from JS ones, if same generic type applied than the following test would need to be modified:
`fastify.assert.equal(1, '1'); fastify.assert.notEqual(1, '2');`

Don't think it's a problem, just pointing it out

Reply:
Actually, I think I'll retract my previous statement. I believe having separate generics would be best here. Keep them all as is!